modules/ezcontent_node/modules/ezcontent_smart_article/src/EzcontentTextToSpeechPluginBase.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Drupal\ezcontent_smart_article;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\media\Entity\Media;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * A base class implementation for text-to-speech conversion plugin manager.
+ */
+abstract class EzcontentTextToSpeechPluginBase extends PluginBase implements EzcontentTextToSpeechInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function convertTextToSpeech($text) {
+  }
+
+  /**
+   * Create and attach audible media to entity.
+   *
+   * @param string $entity_type_id
+   *   Entity Type ID.
+   * @param string $entity_id
+   *   Entity ID.
+   * @param string $field_name
+   *   Field's Machine name in which the created media will be stored.
+   * @param string $speech
+   *   Audio string from text-to-speech service response.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function saveSpeechToEntity($entity_type_id, $entity_id, $field_name, $speech) {
+    $entity = \Drupal::entityTypeManager()
+      ->getStorage($entity_type_id)
+      ->load($entity_id);
+    if ($entity instanceof ContentEntityInterface) {
+      $speech_filename = $entity->bundle() . '-' . $entity->id() . '.mp3';
```

Comment:
Currently, filename can be duplicate. It can be madd unique by adding delta OR Version ID, like `smart_podcast_NID_VID.mp3 / smart_podcast_NID_0/1/2/3.mp3`